src/types.ts

```diff
@@ -16,6 +23,13 @@ export type MeasureEndMessage = {
   msgType: 'measureEnd';
 };
 
+export type PartEndMessage = {
+  msgType: 'partEnd';
+  id: string;
+  ndx: number;
+  from: number;
```

Comment:
nit: Based on its usage, this `from` property references the `ndx`. Would you indicate that in the property name?